File: design-patterns/go/template_method/transport/train.go
// train.go

package main

import "fmt"

type Train struct {
	*Transport
}

func NewTrain() (train *Train) {
	train = &Train{}
	train.Transport = NewTransport(train)
	return
}

func (train *Train) AddEngine() {
	fmt.Println("Adding Engine to Train")
}

func (train *Train) AddWheel() {
	fmt.Println("Adding Wheels to Train")
}

func (train *Train) AddWing() {
	// implementatiom not require for train
}

func (train *Train) CreateBody() {
	fmt.Println("Creating Train Body")
}

File: design-patterns/php/templatemethod/transport/Train.php
<?php
// Train.php

namespace BigBoxCode\DesignPattern\TemplateMethod\Transport;

class Train extends Transport {
    public function createBody(): void {
        echo "Creating Train Body\n";
    }

    public function addEngine(): void {
        echo "Adding Engine to Train\n";
    }

    public function addWheel(): void {
        echo "Adding Wheels to Train\n";
    }

    public function addWing(): void {
        // not required for train
    }
}